src/mem/freelist.h

```diff
@@ -188,6 +173,63 @@ namespace snmalloc
       return n;
     }
 
+  public:
+    template<SNMALLOC_CONCEPT(capptr_bounds::c) B>
+    static CapPtr<FreeObject, B> make(CapPtr<void, B> p)
+    {
+      return p.template as_static<FreeObject>();
+    }
+
+    /**
+     * Assign next_object and update its prev_encoded if SNMALLOC_CHECK_CLIENT.
+     * Static so that it can be used on reference to a FreeObject.
+     *
+     * Returns a pointer to the next_object field of the next parameter as an
+     * optimization for repeated snoc operations (in which
+     * next->next_object is nullptr).
+     */
+    template<SNMALLOC_CONCEPT(capptr_bounds::c) B>
+    static CapPtr<FreeObject, B>* store_next(
+      CapPtr<FreeObject, B>* curr,
+      CapPtr<FreeObject, B> next,
+      const FreeListKey& key)
+    {
+      return reinterpret_cast<CapPtr<FreeObject, B>*>(store_next(
+        reinterpret_cast<FreeObject**>(curr), next.unsafe_ptr(), key));
+    }
+
+    template<SNMALLOC_CONCEPT(capptr_bounds::c) B>
+    static void store_null(CapPtr<FreeObject, B>* curr, const FreeListKey& key)
+    {
+      *curr =
+        CapPtr<FreeObject, B>(encode_next(address_cast(curr), nullptr, key));
+    }
+
+    /**
+     * Assign next_object and update its prev_encoded if SNMALLOC_CHECK_CLIENT
+     *
+     * Uses the atomic view of next, so can be used in the message queues.
+     */
+    template<SNMALLOC_CONCEPT(capptr_bounds::c) B>
+    void atomic_store_next(CapPtr<FreeObject, B> next, const FreeListKey& key)
+    {
+      this->atomic_store_next(next.unsafe_ptr(), key);
+    }
+
+    void atomic_store_null(const FreeListKey& key)
+    {
+      atomic_next_object.store(
+        encode_next(address_cast(&next_object), nullptr, key),
+        std::memory_order_relaxed);
+    }
+
+    template<SNMALLOC_CONCEPT(capptr_bounds::c) B>
+    static CapPtr<FreeObject, B>
+    atomic_read_next(CapPtr<FreeObject, B> self, const FreeListKey& key)
```

Comment:
Why has this become a static method (and why is it wrapping a non-static method?)